### PR TITLE
Update github tests-functional-ubuntu actions

### PR DESCRIPTION
**Pull request by mcmilk**
Some changes to the Ubuntu functional testings.

### Motivation and Context

On my local machines, the tests give some more failings and I wondered why.
On the Github Action VMs with Ubuntu version 22.04 these Tests fail also... so this happens only to newer libs / kernels.

### Description

This commit changes the tests-functional-ubuntu actions:
- add Ubuntu 22.04 to the build list
- report info about CPU and this file: /proc/spl/kstat/zfs/chksum_bench
- increase timeout for the tests to 6 hours, cause the Ubuntu 18.04 machines are sometimes a bit slower and need more time now

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)
- [x] Ubuntu Testing suite

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by andrewc12** [comment]
We've also been running into this timeout issue in a few of our pull requests.
@behlendorf I beleive you're mostly in charge of the ci so you might be the one to look at this.

On a side note should we possibly look at some way of splitting the runfiles up into seperate groups of tests that we run concurrently on different machines?

**Comment by rincebrain** [comment]
So, I believe I set 330m because the default timeout for the whole action is 360m, and if that gets hit, it doesn't run any of the debug capture and cleanup actions, you just wind up with "oh the workflow died :(".

So setting the timeout to 360m in that step, I believe, guarantees you'll never hit it and also won't collect debug artifacts when the whole action dies.

My other remark, about parallelizing ZTS runs, would be that there have been both historical and current bugs that only trigger if you run a bunch of the tests in one run, so we may find that we get a codebase that finishes tests in 5 minutes and doesn't actually pass if you ran them in full on one instance...

**Comment by andrewc12** [comment]
This is a bit of a problem.

Maybe we will have to do something like have the github runners execute in parallel and the build bot run the full set of tests.

I was considering splitting them into seperate tests anyway to try to speed up the dev test cycle for the macos and windows branches.

**Comment by andrewc12** [comment]
I'm trying out a naive attempt here
https://github.com/andrewc12/openzfs/actions/runs/3264412810

**Comment by gmelikov** [comment]
@andrewc12 I've already tried naive way with Travis CI https://github.com/openzfs/zfs/pull/6829/files , it's really ugly and manual. But Travis has really tight 60min timeouts, and in Github's case it's ok to just split tests by count automatically. And looks like there is a way to make matrix dynamic https://tomasvotruba.com/blog/2020/11/16/how-to-make-dynamic-matrix-in-github-actions/ .

**Comment by andrewc12** [comment]
I wondering if I could just write a script that adds n number of values to the matrix and then when they run it splits it into n files and picks the corresponding file.

**Comment by behlendorf** [comment]
My only real concern is increasing the 330s time limit, as mentioned above, but after re-reading the [limits documentation](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration) it sounds like that 6 hour limit is for the job and not the entire workflow.  So perhaps it wouldn't be a problem and we'd still be able to save any artifacts. 

    Job execution time - Each job in a workflow can run for up to 6 hours of
    execution time. If a job reaches this limit, the job isterminated and fails
    to complete.

    Workflow run time - Each workflow run is limited to 35 days. If a workflow
    run reaches this limit, the workflow run is cancelled. This period includes
    execution duration, and time spent on waiting and approval.

**Comment by rincebrain** [comment]
I'm a little confused why my change helped that problem at all, then, unless there's also a larger default timeout or I'm forgetting some detail about that commit.

**Comment by andrewc12** [comment]
Sometimes the builds seem to run faster.

We could squeeze some extra time out by having two jobs in the workflow.
1st that builds zfs and uploads it as an artifact.
A 2nd one depends on the first that installs from that artifact and then runs the tests.

For example see
https://github.com/andrewc12/openzfs/blob/andrew_workflow-01/.github/workflows/windows-build-test.yml

**Comment by behlendorf** [comment]
> Sometimes the builds seem to run faster.

I've noticed this as well.  What's not clear to me is if in those cases if an extra 15 or 30 minutes would be enough.

It also appears we have a group of tests failing on the new Ubuntu 22.04 builder, 2/2 runs so far.  I'll resubmit it again to make sure it's reproducible.  We'll need to run whatever is causing that down before adding that builder.

**Comment by mcmilk** [comment]
> I've noticed this as well. What's not clear to me is if in those cases if an extra 15 or 30 minutes would be enough.

I increased the timeout to 340 minutes and downgraded the `3G` to `2G` ... so maybe the raidz/draid testings will be a bit faster now.

The CPU Information tells us some differences of the github machines in the current runnings....
- Ubuntu 18, 8272CL, MAKE => 9m 46s
- Ubuntu 20, E5-2673 v4, MAKE => 14m 17s
- Ubuntu 22, E5-2673 v3, MAKE => 13m 1s

> It also appears we have a group of tests failing on the new Ubuntu 22.04 builder, 2/2 runs so far. I'll resubmit it again to make > sure it's reproducible. We'll need to run whatever is causing that down before adding that builder.

Yes, these tests are also failing on my local test boxes and were the reason for creating this PR.

**Comment by gmelikov** [comment]
While patch is LGTM, we may want to fix failing tests on 22.04 or add them to known errors before merge (worst way), otherwise other PRs will see false errors without dependence to PRs' changes.

**Comment by behlendorf** [comment]
It seems the GitHub 18.04 runners have also been deprecated and are scheduled to be retired in April 2023.  Leading up to this there will also be intentional outages/failures to draw attention to their upcoming removal.  

https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/

Instead of extending the timeout for the 18.04 hardware, let's go ahead and retire it and focus on getting a clean test run for a replacement 22.04 runner.

File: .github/workflows/zfs-tests-functional.yml [hunk 1]
```diff
@@ -54,15 +54,19 @@ jobs:
...
+        lscpu
+        cat /proc/spl/kstat/zfs/chksum_bench
     - name: Report disk space
       run: |
         df -h /
     - name: Tests
       run: |
         set -o pipefail
-        /usr/share/zfs/zfs-tests.sh -vKR -s 3G | scripts/zfs-tests-color.sh
+        /usr/share/zfs/zfs-tests.sh -vKR -s 2G | scripts/zfs-tests-color.sh
```
Comment (behlendorf) [inline .github/workflows/zfs-tests-functional.yml:67]: This change is responsible for some of the new test failures.  We've got a handful of tests which expect a slightly larger minimum pool size.

Comment (mcmilk) [reply .github/workflows/zfs-tests-functional.yml:67]: okay, I will revert this :(

**Comment by szubersk** [comment]
ZTS on Ubuntu 22.04 fails consistently for me too.
I'd suggest starting by switching zloop and sanity tests to Ubuntu 22.04 as it'd be much easier and already gives the project coverage on 22.04.
As Brian wrote, Ubuntu has 5 years expiration date, and focusing on 18.04 is a waste of time.

**Comment by mcmilk** [comment]
The PR: https://github.com/openzfs/zfs/pull/14078 obsoletes this one.

**Pull request closed without merging**